### PR TITLE
contrib/libarchive: Add illumos support

### PR DESCRIPTION
**Pull request by citrus-it**
This change adds the files necessary to build contrib/libarchive
on illumos.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: contrib/libarchive-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -135,10 +135,18 @@ set(SRCS
...
 add_library(_libarchive ${SRCS})
-target_include_directories(_libarchive PUBLIC 
-    ${CMAKE_CURRENT_SOURCE_DIR}
-    "${LIBRARY_DIR}/libarchive"
-)
+if (OS_SUNOS)
+	target_include_directories(_libarchive PUBLIC
+	    ${CMAKE_CURRENT_SOURCE_DIR}/illumos
+	    "${LIBRARY_DIR}/libarchive"
+	    "/usr/include/libxml2"
```
Comment (alexey-milovidov) [inline contrib/libarchive-cmake/CMakeLists.txt:142]: This won't work for cross-compilation and for hermetic builds. Let's find where libxml2 is inside our source tree...

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 9180f3f6da01c88b5e73aa52cdc766bce9799b30 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61703/9180f3f6da01c88b5e73aa52cdc766bce9799b30/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: contrib/libarchive-cmake/CMakeLists.txt (hunk 1 above)
Comment (citrus-it) [reply contrib/libarchive-cmake/CMakeLists.txt:142]: Ah yes, I'll come back to this one, thanks.

**Comment by alexey-milovidov** [comment]
No activity.

**Pull request closed without merging**